packages/location-field/src/index.js

```diff
@@ -356,38 +418,51 @@ class LocationField extends Component {
       // Add the menu sub-heading (not a selectable item)
       // menuItems.push(<MenuItem header key='sr-header'>Search Results</MenuItem>)
 
+      // Split out transit results
+      const stopFeatures = geocodedFeatures
+        .filter(feature => feature.properties.layer === "stops")
+        .slice(0, 3);
+      const stationFeatures = geocodedFeatures
+        .filter(feature => feature.properties.layer === "stations")
+        .slice(0, 3);
+      const otherFeatures = geocodedFeatures
+        .filter(feature => feature.properties.source !== "transit")
+        .slice(0, 3);
+
       // Iterate through the geocoder results
       menuItems = menuItems.concat(
-        geocodedFeatures.map(feature => {
-          // Create the selection handler
-          const locationSelected = () => {
-            getGeocoder(geocoderConfig)
-              .getLocationFromGeocodedFeature(feature)
-              .then(geocodedLocation => {
-                // Set the current location
-                this.setLocation(geocodedLocation, "GEOCODE");
-                // Add to the location search history. This is intended to
-                // populate the sessionSearches array.
-                addLocationSearch({ location: geocodedLocation });
-              });
-          };
-
-          // Add to the selection handler lookup (for use in onKeyDown)
-          this.locationSelectedLookup[itemIndex] = locationSelected;
-
-          // Create and return the option menu item
-          const option = (
-            <Option
-              icon={<GeocodedOptionIconComponent feature={feature} />}
-              key={optionKey++}
-              title={feature.properties.label}
-              onClick={locationSelected}
-              isActive={itemIndex === activeIndex}
-            />
-          );
-          itemIndex++;
-          return option;
-        })
+        stationFeatures.length > 0 && (
+          <Styled.MenuItem
+            color={layerColorMap.stations}
+            centeredText
+            header
+            key="gtfs-stations-header"
+          >
+            Stations
+          </Styled.MenuItem>
+        ),
+        stationFeatures.map(feature =>
+          this.renderFeature(itemIndex++, feature)
+        ),
+
+        stopFeatures.length > 0 && (
+          <Styled.MenuItem
+            color={layerColorMap.stops}
+            centeredText
+            header
+            key="gtfs-stops-header"
```

Comment:
Sort props

Reply:
Sorted in e0fc8a811358ed8a55a587f948db1614da68dbd4